src/samples.c

```diff
@@ -0,0 +1,156 @@
+/* SPDX-License-Identifier: BSD-2-Clause
+ *
+ * This file is part of libtslitex.
+ * Access the Sample Data stream from the Thunderscope
+ * LiteX design
+ *
+ * Copyright (C) 2024 / Nate Meyer  / nate.devel@gmail.com
+ *
+ */
+
+
+
+#if defined(_WIN32)
+#include <Windows.h>
+#define ssize_t int64_t
+#else
+#include <unistd.h>
+#include <sys/mman.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+
+#include "samples.h"
+#include "ts_common.h"
+
+#include "liblitepcie.h"
+
+
+#if defined(_WIN32)
+#define TS_DMA_NAME         "\\DMA%u%u"
+#define TS_DMA_NAME_LEN     (16)
+#define TS_DMA_NAME_ARGS(chan, dev)     ((chan), (dev))
+#define TS_DMA_OS_FLAGS     (FILE_ATTRIBUTE_NORMAL | \
+                             FILE_FLAG_NO_BUFFERING)
+#else
+#define INVALID_HANDLE_VALUE (-1)
+#define TS_DMA_NAME         "/dev/litepcie%u"
+#define TS_DMA_NAME_LEN     (24)
+#define TS_DMA_NAME_ARGS(chan, dev)     ((dev))
+#define TS_DMA_OS_FLAGS     (O_RDWR | O_CLOEXEC)
+#endif
+
+
+int32_t samples_init(sampleStream_t* inst, uint8_t devIdx, uint8_t channel)
+{
+    int32_t retVal = TS_STATUS_ERROR;
+    char devName[TS_DMA_NAME_LEN] = {0};
+    
+    if(inst)
+    {
+        inst->dma_buffer_count = 0;
+        inst->driver_buffer_count = 0;
+        inst->active = 0;
+
+        snprintf(devName, TS_DMA_NAME_LEN, TS_DMA_NAME, TS_DMA_NAME_ARGS(channel, devIdx));
+
+        inst->dma = litepcie_open(devName, TS_DMA_OS_FLAGS);
+        
+        if((INVALID_HANDLE_VALUE != inst->dma) &&
+            litepcie_request_dma(inst->dma, 0, 1))
+        {
+            litepcie_dma_set_loopback(inst->dma, 0);
+            retVal = TS_STATUS_OK;
+        }
+        //else, DMA Unavailable
+    }
+
+    return retVal;
+}
+
+int32_t samples_enable_set(sampleStream_t* inst, uint8_t en)
+{
+    if(NULL == inst)
+    {
+        return TS_STATUS_ERROR;
+    }
+
+    inst->active = en;
+    
+    //Enable Trigger
+    litepcie_writel(inst->dma, CSR_ADC_TRIGGER_CONTROL_ADDR, en);
+
+    //Start/Stop DMA
+    litepcie_dma_writer(inst->dma, en,
+                        &inst->dma_buffer_count,
+                        &inst->driver_buffer_count);
+
+    return TS_STATUS_OK;
+}
+
+int32_t samples_get_buffers(sampleStream_t* inst, uint8_t* sampleBuffer, uint32_t bufferLen)
```

Comment:
I think it would be good to document these functions and the meaning of the parameters and especially the return value

Reply:
Function documentation is in the header file. Let me know if you want any more detail for any of them.